restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -649,6 +653,134 @@ private void info(final SipServletRequest request) throws IOException {
+
+        ActorRef relatedCall = null;
+        List<ActorRef> listOfRelatedCalls = null;
+        if (answer instanceof ActorRef) {
+            relatedCall = (ActorRef) answer;
+        } else if (answer instanceof List) {
+            listOfRelatedCalls = (List<ActorRef>) answer;
+        }
+
+        if(logger.isInfoEnabled()) {
+            logger.info("About to start Live Call Modification");
+            logger.info("Initial Call path: " + call.path());
+            if (relatedCall != null) {
+                logger.info("Related Call path: " + relatedCall.path());
+            }
+            if (listOfRelatedCalls != null) {
+                logger.info("List of related calls received, size of the list: "+listOfRelatedCalls.size());
+            }
+            // Cleanup all observers from both call legs
+            logger.info("Will tell Call actors to stop observing existing Interpreters");
+        }
+        if (logger.isDebugEnabled()) {
+            logger.debug("LCM account: "+ cdr.getAccountSid() +", new RCML url: "+ number.getReferUrl());
+        }
+        call.tell(new StopObserving(), self());
+        if (relatedCall != null) {
+            relatedCall.tell(new StopObserving(), self());
+        }
+        if (listOfRelatedCalls != null) {
+            for(ActorRef branch: listOfRelatedCalls) {
+                branch.tell(new StopObserving(), self());
+            }
+        }
+        if(logger.isInfoEnabled()) {
+            logger.info("Existing observers removed from Calls actors");
+
+            // Cleanup existing Interpreter
+            logger.info("Existing Interpreter path: " + existingInterpreter.path() + " will be stopped");
+        }
+        existingInterpreter.tell(new StopInterpreter(true), null);
+
+        // Build a new VoiceInterpreter
+        final VoiceInterpreterBuilder builder = new VoiceInterpreterBuilder(system);
+        builder.setConfiguration(configuration);
+        builder.setStorage(storage);
+        builder.setCallManager(self());
+        builder.setConferenceManager(conferences);
+        builder.setBridgeManager(bridges);
+        builder.setSmsService(sms);
+        builder.setAccount(cdr.getAccountSid());
+        builder.setVersion(cdr.getApiVersion());
+        builder.setMethod(number.getReferMethod());
+
+        if (number.getReferApplicationSid() != null) {
+            Application application = storage.getApplicationsDao().getApplication(number.getReferApplicationSid());
+            builder.setReferUrl(UriUtils.resolve(application.getRcmlUrl()));
+        } else {
+            builder.setReferUrl(UriUtils.resolve(number.getReferUrl()));
+        }
+
+        builder.setReferMethod((number.getReferMethod() != null && number.getReferMethod().length() > 0) ? number.getReferMethod() : "POST");
+        builder.setReferTarget(request.getHeader("Refer-To").replace("<", "").replace(">", ""));
```

Comment:
@vmykhailiuk better don't parse string, prefer using SIP URI like : 
`((SipURI)request.getAddressHeader("Refer-To").getURI()).getUser()`